pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+The repository allows use of multiple namespaces with the default being the
+standard pywmbe default namespace.
+
+This mocker uses the same APIs as the standard WBEMConnections (it actually
+uses the methods and only mocks the communication with the WBEMServer) and
+returns syntatically the same forms of data and exceptions.
+
+- Class methods:
+
+The methods that get data require the expected classes to be in the repository
+before the mock call. The methods are:
+
+  - GetClass - Gets a single class from a namespace in the repository
+  - EnumerateClasses - Gets list of classes from a namespace in the
+    repository
+  - EnumerateClassNames - Gets list of class names from the repository
+  - CreateClass - Creates a class in the repository
+  - DeleteClass - Deletes a class in the repository
+
+-CIMQualifierDeclaration methods:
+
+  - SetQualifier - Sets a new QualifierDeclaration into a namespace in the
+    repository
+  - GetQualifier - Gets a QualifierDeclaration from the repository
+  - EnumerateQualifiers - Gets all QualifierDeclaration entities from a
+    namespace in the repository
+
+-CIMInstance methods
+
+The methods that get data require both classes and instances in the repository
+for the instances to be recovered.
+
+  - GetInstance - Gets an instance from a namespace in the repository
+  - EnumerateInstances - Gets instances for a class defined in a namespace
+    in the repository.  If there is no class repository, it gets instances
+    for the defined class.  If a class definition for the target class is in
+    the repository, it attempts to get instances of this class and all
+    subclasses
+  - EnumerateInstanceNames - Gets instance names for a class defined in
+    a namespace in the repository.  If there is no class repository, it gets
+    names or the defined class.  If a class definition for the target class is
+    in the repository, it attempts to get instances of this class and all
+    subclasses
+  - CreateInstance - TODO
+  - ModifyInstance - TODO
+  - ExecQuery - This method will return fixed data based on the query string.
+    It does not execute the select statement itself nor does it parse it.
+    An incorrect select statement will be ignored.
+
+- Associators and References
+ - AssociatorNames
+ - Associators
+ - ReferenceNames
+ - References
+
+Supports both class and instance methods
+
+-InvokeMethod
+
+- Pull Operaions - All of the Open and Pull methods are included. However,
+  any Query Language statements are ignored and the full instance responses
+  generated. Since FQL only filters the set of instanced defined by
+  the corresponding request (i.e. Open/Pull EnumerateInstances simply
+  filters instances that would be acquired by EnumerateInstances. This means
+  that these methods can be used but may return more instances than would
+  be returned without the filters.
+
+These methods require the required classes and instances to be in the repository
+for what is expected to be returned
```

Comment:
This statement is no longer needed when we make a more comprehensive statement before the list of operations (see comment above).

Reply:
See above comment.